src/spikeinterface/core/baserecording.py

```diff
@@ -331,6 +331,11 @@ def get_traces(
         segment_index = self._check_segment_index(segment_index)
         channel_indices = self.ids_to_indices(channel_ids, prefer_slice=True)
         rs = self._recording_segments[segment_index]
+        start_frame = int(start_frame) if start_frame is not None else 0
+        num_samples = rs.get_num_samples()
+        end_frame = int(min(end_frame, num_samples)) if end_frame is not None else num_samples
+        if start_frame < 0:
+            raise ValueError("start_frame cannot be negative")
```

Comment:
Why having a clip for upper bounds and making a raise for lower bounds ?
we could also have a max(0, start_frame) similar to min(end_frame, num_samples)
no ?

Reply:
we decided to raise errors in #1989

Reply:
And you voted for exaclty that against me 💔 : (

Reply:
me???

Reply:
I meant Sam : )

Reply:
So we need also to raise error here also no ?

Reply:
for upper bound I mean